### ISSUE TITLE
Suggestion for how to make this module safer

### ISSUE THREAD

**Issue by dmfenton** [agershun/alasql#662]
I came across an interesting module in the Node stdlib called vm. Since this library makes heavy use of evaluating strings as functions as I though this may be something to look at.

https://nodejs.org/api/vm.html

It allows you to specify to explicitly specify the execution context for a given script. Therefore you can disable things like requiring external modules, the filesystem api, etc

```
const vm = require('vm')
vm.runInContext('require("foo")', {})
ReferenceError: require is not defined
```

**Comment by mathiasrw** [comment]
So you would use vm to run the string constructed functions?

It would be interesting to see if there is a performance gain or loss from this. 

On another note: would also be usefull to [run alasql as a whole in a vm](https://github.com/agershun/alasql/issues/451)

**Comment by dmfenton** [comment]
I agree the performance trade off would be worth measuring. I don't think you could run all of alasql inside a vm without disabling user defined functions. User defined functions in general are the tricky part I think.

**Comment by mathiasrw** [comment]
Do you know if vm can run within a vm?

**Comment by dmfenton** [comment]
```
> vm.runInContext(vm.runInContext('2*2', vm.createContext()), vm.createContext({vm}))
4
```
